custom/text-chat/contexts/ChatProvider.js

```diff
@@ -6,61 +6,73 @@ import React, {
   useState,
 } from 'react';
 import { useCallState } from '@custom/shared/contexts/CallProvider';
+import { useSharedState } from '@custom/shared/hooks/useSharedState';
 import { nanoid } from 'nanoid';
 import PropTypes from 'prop-types';
 
 export const ChatContext = createContext();
 
 export const ChatProvider = ({ children }) => {
   const { callObject } = useCallState();
-  const [chatHistory, setChatHistory] = useState([]);
+  const { sharedState, setSharedState } = useSharedState({
+    initialValues: {
+      chatHistory: [],
+    },
+    broadcast: false,
```

Comment:
Does this mean the broadcast functionality is not actually used?

Reply:
yep, for the current use-case, we are anyway handling the state through the handleNewMessage, so we can set the broadcast to false.

In this case (broadcast = false) the shared-state works only when a user requests for the state.